service/handlers.go

```diff
@@ -3,22 +3,45 @@ package service
 import (
 	"encoding/json"
 	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/go-semver/semver"
 	"github.com/gorilla/mux"
 	"github.com/rancher/go-rancher/client"
 	"github.com/rancher/rancher-catalog-service/manager"
 	"github.com/rancher/rancher-catalog-service/model"
 	"net/http"
+	"regexp"
 )
 
 const headerForwardedProto string = "X-Forwarded-Proto"
 
 //ListTemplates is a handler for route /templates and returns a collection of template metadata
 func ListTemplates(w http.ResponseWriter, r *http.Request) {
-	//read the catalog
 
-	log.Debug("Request to list Templates Catalog")
+	rancherVersion := r.URL.Query().Get("minimumRancherVersion_lte")
+	if rancherVersion != "" {
+		log.Debugf("Request to get all templates with minimumRancherVersion <= %s", rancherVersion)
+	} else {
+		log.Debug("Request to list All Templates in the Catalog")
+	}
+
+	//read the catalog
 	resp := model.TemplateCollection{}
 	for _, value := range manager.Catalog {
+
+		if rancherVersion != "" {
+			var err error
+			value.VersionLinks, err = filterByMinimumRancherVersion(rancherVersion, &value)
+			if err != nil {
+				//cannot apply the filter, return empty set
+				break
+			}
+		}
+
+		//if no versions are present then just skip the template
+		if len(value.VersionLinks) == 0 {
```

Comment:
Here, in case this means that if a template has no subversions (like a folder 0, 1,2 etc) at all under it, then that template is skipped. This is not related to skipping the template is there is no minimum rancher version